lib/galaxy/jobs/runners/drmaa.py

```diff
@@ -224,6 +224,8 @@ def queue_job(self, job_wrapper):
             if external_job_id is None:
                 job_wrapper.fail(f"({galaxy_id_tag}) could not queue job")
                 return
+            else:
+                os.unlink(filename)
```

Comment:
How about storing this in the job wrapper directory? that way you'll benefit from the various cleanup_after_job settings ?

Reply:
This was exactly what I just though about (see my edit of the original message). Might also render the config variable `cluster_files_directory` obsolete if we also get rid of it in the pbs runner (its used to store .o .e and .sh files ... shouldn't those go to the jwd as well).

Even if this does not help me with a current problem (for which unlinking is a workaround)

The problem is that on resubmission the template is overwritten .. but on our system opening the template for the 2nd submission sometimes does not truncate its content which creates invalid JSON. Our admins are investigating. But if you or somebody else has an idea it would be welcome, I'm guessing for a buffering / filesystem problem.